### DISCUSSION TITLE
How to avoid flickering when pushSprite over another sprite

### DISCUSSION THREAD

**Discussion by iweave1** [Bodmer/TFT_eSPI#2732] [Q&A - General]
I am trying to animate a small game on TTGO T-Display S3 with 6 pushbuttons. I have a sprite for the background, an sprite for the player character. When I update the player position and push it to TFT it causes flashing. It did not do this when I used the same image each push over the background sprite.
How can I animate a moving sprite?

I am using Arduino IDE.
 2.5.0 version od TFT_eSPI.
Board package version 2.0.9
ESP32 S3 TTGO T-Display
 ST7789v TFT Driver
8 bit Parrallel interface
[TestGame.tar.gz](https://github.com/Bodmer/TFT_eSPI/files/11951770/TestGame.tar.gz)

**Comment by Bodmer** [comment]
Flickering will occur if the pixel colours are changed rapidly on the display.

I have not written games but the project videos here might help:
https://www.youtube.com/@VolosProjects

Ideally a full frame buffer (aka large sprite) is needed and this must be completely updated before writing to the screen.  For large screens the update rate of the complete display area will limit the maximum farmes rate.
